File: app/src/main/java/cn/com/woong/readhub/ui/topic/TopicPresenter.java
package cn.com.woong.readhub.ui.topic;

import javax.inject.Inject;
import cn.com.woong.readhub.App;
import cn.com.woong.readhub.base.BasePresenter;
import cn.com.woong.readhub.constant.Constant;
import cn.com.woong.readhub.domain.apiservice.ReadhubApiService;
import cn.com.woong.readhub.domain.RxSchedulers;
import cn.com.woong.readhub.resp.TopicResp;
import io.reactivex.functions.Consumer;

/**
 * Created by wong on 2018/3/9.
 */

public class TopicPresenter extends BasePresenter<TopicContract.View> implements TopicContract.Presenter {
//    @Inject
    public TopicPresenter() {
    }

    @Override
    public void getTopicNews(final String order) {
        App.Companion.apiService(ReadhubApiService.class)
                .apiTopic(order, Constant.TOPIC_PAGE_SIZE)
                .compose(RxSchedulers.<TopicResp>io_main())
                .compose(mView.<TopicResp>bindToLife())
                .subscribe(new Consumer<TopicResp>() {
                    @Override
                    public void accept(TopicResp topicResp) throws Exception {
                        if (topicResp != null && topicResp.data != null) {
                            mView.updateTopicData(order, topicResp.data);
                        }
                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {
                        mView.showFailed();
                    }
                });
    }
}

File: app/src/main/java/cn/com/woong/readhub/ui/topic/TopicPresenter.kt
package cn.com.woong.readhub.ui.topic

import android.R.attr.data
import android.annotation.SuppressLint
import cn.com.woong.readhub.network.resp.TopicResp
import cn.com.woong.readhub.network.RxSchedulers
import cn.com.woong.readhub.network.ReadhubApiService
import cn.com.woong.readhub.App
import cn.com.woong.readhub.base.BasePresenter
import cn.com.woong.readhub.constant.Constant
import com.blankj.utilcode.util.LogUtils
import io.reactivex.functions.Consumer


class TopicPresenter : BasePresenter<TopicContract.View>(), TopicContract.Presenter {

    @SuppressLint("CheckResult")
    override fun getTopicNews(order: String) {
        App.sInstance.apiService(ReadhubApiService::class.java)
                ?.apiTopic(order, Constant.TOPIC_PAGE_SIZE)
                ?.compose(RxSchedulers.io_main())
                ?.subscribe(Consumer<TopicResp> { topicResp ->
                    if (topicResp.data != null) {
                        mView?.updateTopicData(order, topicResp.data!!)
                    }
                }, Consumer<Throwable> {
                    LogUtils.e("apiTopic error ==== ${it}")
                })
    }
}